src/Functions/registerFunctionsIntrospection.cpp

```diff
@@ -1,27 +0,0 @@
-namespace DB
-{
-
-class FunctionFactory;
-
-#if defined(OS_LINUX)
```

Comment:
Here, we previously skipped the registration of `registerFunctionAddressToSymbol()` if we are not on Linux (`OS_LINUX`). If you look in "Functions/addressToSymbol.cpp", the corresponding ifdef is `#if defined(__ELF__) && !defined(OS_FREEBSD)`. For ClickHouse's current supported platforms, that is equivalent to Linux, but perhaps that might change one fine day.

So my question would be if you like to check for all functions which are likewise compiled only conditionally (e.g. based on OS or enabled/disabled 3rd party dependency), whether there are similar subtleties and make these consistent?

Reply:
OK will do

Reply:
@rschu1ze Here's the list, please let me know if any action is needed.
## registerFunctionAddressToXXX 
```cpp
#if defined(OS_LINUX)
    registerFunctionAddressToSymbol(factory);
    registerFunctionAddressToLine(factory);
    registerFunctionAddressToLineWithInlines(factory);
#endif
```
`#if defined(__ELF__) && !defined(OS_FREEBSD)` is used in implementation, which is matching the definition is `SymbolIndex.h`. 

## serverConstants.cpp
`#if defined(OS_LINUX)` is used to enclose `FunctionGetOSKernelVersion`, but seems the restriction can be relaxed?

## registerFunctionShowCertificate
```
#if USE_SSL
void registerFunctionShowCertificate(FunctionFactory &);
#endif
```
Implementation is not enclosed with USE_SSL, however function will throw exception so it's fine.

## trap.cpp
The function is actually disabled, might just remove this file?